Introduce single stack IPv6 documentation (#6318)
k0sproject/k0s: docs/single-stack-ipv6.md

<!--
SPDX-FileCopyrightText: 2025 k0s authors
SPDX-License-Identifier: CC-BY-SA-4.0
-->

# IPv6 single-stack networking

> K0s introduced experimental IPv6 single-stack networking, allowing your cluster to operate

Reviewer: ```suggestion
K0s introduced experimental IPv6 single-stack networking, allowing clusters to operate
```

exclusively with IPv6 addresses. IPv6 single stack is currently in alpha state and requires

> to enable explicitly the [IPv6SingleStack feature gate](feature-gates.md) on every controller.

Reviewer: ```suggestion
the [IPv6SingleStack feature gate](feature-gates.md) to be enabled on every controller.
```

IPv6 single-stack is fully compatible with both [node-local load balancing](nllb.md) and
[control plane load balancing (CPLB)](cplb.md).

## Enabling IPv6 single-stack using the default CNI (Kube-router)

In order to enable IPv6 single-stack networking using the default CNI provider both an IPv6
`podCIDR` and `serviceCIDR` must be provided. dual-stack must not be enabled. Migrations

> to dual-stack are possible as long as the primary address family remains IPv6.

Reviewer: ```suggestion
to dual stack are possible as long as the primary address family remains IPv6.
```

```yaml
spec:
  network:
    podCIDR: fd00::/108
    serviceCIDR: fd01::/108
```

Ensure that every controller in the cluster is executed with the
[IPv6SingleStack feature gate](feature-gates.md) enabled:

```shell
k0s controller --feature-gates="IPv6SingleStack=true"
```

This configuration will set up all Kubernetes components and Kube-router for IPv6
single-stack networking.

> ### Configuring the node CIDR mask size

Reviewer: Given that this section is about to be added in #6319 to the dual-stack docs, as well, I wonder if we should try to merge these doc pages, maybe? Or have a more organized structure in the docs? Say, like this:

```yaml
      - Networking (CNI):
          - Overview: networking.md
          - IPv4/IPv6 dual stack: dual-stack.md
          - IPv6 single stack: single-stack-ipv6.md
```

Author: Makes perfect sense, I'll address this in a third PR
